### PR TITLE
Split Tickets page into 4 sub tabs

### LINKED ISSUE #813: Split Tickets Page into Sub-menu/Tabbed Pages
The current tickets page should be split into the 4 future sub-menu/tabbed page as requested by @linnutee (as seen here: 2-4  [mockup](https://xd.adobe.com/view/7707b586-62f0-4c93-afaf-1c5b9432b8eb/screen/97287f68-fc56-4c15-a903-f60a29722595/My-Tickets-2)

Purchase Tickets, My Tickets, Governance, Statistics

Purchase Tickets should be the current ticket purchasing view.
Governance should be the current voting view.

My Tickets and Statistic should display "Coming soon" or some other stubbed text.

### PR DESCRIPTION
**Pull request by alexlyp**
Closes #813, #814 
This PR completes the change over to using the header menu and tab selection.  This also introduces a space for the upcoming My Tickets and Statistics pages.  

Other changes that are included:
- Allow users to see the governance/voting choice tab without having a stakepool set up.
- Using the new snackbar notifications for ticket page success/errors.

Issues remaining to be fixed for this PR: 
- ~Proper transitions between Ticket tabs.~
- ~Update copy for placeholders and Tickets header~
- ~Erroneous stop ticket buyer failure on startup.  Because we issue a StopAutoBuyerRequest on startup an error in the new snack bar notification is being shown, but we need to hide that somehow.~ Will open an issue seperately for this.

### CODE REVIEW

File: app/components/views/TicketsPage/GovernanceTab/index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import React from "react";
 import { autobind } from "core-decorators";
 import VotingPrefsPage from "./Page";
-import votingPrefs from "../../connectors/votingPrefs";
-import { find, compose, eq, get, substruct } from "../../fp";
+import votingPrefs from "../../../../connectors/votingPrefs";
+import { find, compose, eq, get, substruct } from "../../../../fp";
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/GovernanceTab/index.js]: #779 obviates the need for this.

You can just do an absolute path from "app" or "components" now.

So lines 4 and 5 can just be
```
import { votingPrefs } from "connectors";
import { find, compose, eq, get, substruct } from "fp";
```

File: app/components/views/TicketsPage/Header.js [hunk 2]
```diff
@@ -1,141 +1,94 @@
...
+  "tickets.title":               { id: "tickets.title",               defaultMessage: "Tickets" },
+  "tickets.description":         { id: "tickets.description",         defaultMessage: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua." },
+  "tickets.tab.purchase":        { id: "tickets.tab.purchase",        defaultMessage: "Purchase Tickets" },
+  "tickets.tab.mytickets":       { id: "tickets.tab.mytickets",       defaultMessage: "My Tickets" },
+  "tickets.tab.governance":      { id: "tickets.tab.governance",      defaultMessage: "Governance" },
+  "tickets.tab.statistics":      { id: "tickets.tab.statistics",       defaultMessage: "Statistics" },
+});
+
+@autobind
+class Header extends React.Component {
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/Header.js]: This thing could realistically be pulled out into it's own component instead of being defined in both tickets and transactions.  Maybe that should be a separate PR?

File: app/components/views/TicketsPage/MyTicketsTab/Page.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import "style/Layout.less";
+import "style/ReceivePage.less";
+import "style/MiscComponents.less";
+
+const MyTicketsPage = () => (
+  <div className="tab-card">
+    <div className="receive-content-nest">
+      My Tickets Page coming soon!
+    </div>
+    <ReactTooltip />
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/MyTicketsTab/Page.js]: lol delet dis

File: app/components/views/TicketsPage/Header.js (hunk 2 above)
Comment (alexlyp) [reply app/components/views/TicketsPage/Header.js]: Yah that was my thought as well.  We can do in a seperate PR

File: app/components/views/TicketsPage/MyTicketsTab/Page.js (hunk 3 above)
Comment (alexlyp) [reply app/components/views/TicketsPage/MyTicketsTab/Page.js]: Yup more cleaning up.

File: app/components/views/TicketsPage/PurchaseTab/Tickets.js [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
-import TicketAutoBuyer from "../../TicketAutoBuyer";
-import PassphraseModal from "../../PassphraseModal";
-import ImportScriptModal from "../../ImportScriptModal";
+import PurchaseTickets from "../../../PurchaseTickets";
+import TicketAutoBuyer from "../../../TicketAutoBuyer";
+import PassphraseModal from "../../../PassphraseModal";
+import ImportScriptModal from "../../../ImportScriptModal";
 import { FormattedMessage as T } from "react-intl";
-import StakeInfo from "../../StakeInfo";
+import StakeInfo from "../../../StakeInfo";
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/PurchaseTab/Tickets.js]: here, too. You can remove all the relative imports.

File: app/components/views/TicketsPage/StatisticsTab/Page.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import "style/Layout.less";
+import "style/ReceivePage.less";
+import "style/MiscComponents.less";
+
+const MyTicketsPage = () => (
+  <div className="tab-card">
+    <div className="receive-content-nest">
+      Statistics Page
+    </div>
+    <ReactTooltip />
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/StatisticsTab/Page.js]: another tooltip slipped through

File: app/components/views/TicketsPage/index.js [hunk 6]
```diff
@@ -1,183 +1,78 @@
...
+const opts = { stiffness: 150, damping: 20 };
+
+const wrapperComponent = props => <div className="tab-content" { ...props } />;
+
+const Tickets = ({ children, location }) => {
+  const tabs = ["purchase", "mytickets", "governance", "statistics"];
+  const page = "tickets";
+  // this will be removed w/ react router 4
+  const pathname = location.pathname.split("/")[2];
+  const effect = pathname === "purchase" ? enterLeft : enterRight;
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/index.js]: if the pathname is tabs[0], enterLeft.
if the pathname is tabs[tabs.length -1], enter right.

if the pathname is anything in between those, use location.history to check what the previous one was (using the same .split("/")[2] ).
if tabs.indexOf(prev) >tabs.indexOf(pathname), enterLeft, else enter right.

Comment (alexlyp) [reply app/components/views/TicketsPage/index.js]: cool thx.  was missing the location.history piece for this.

Comment (Jascha-Sundaresan) [reply app/components/views/TicketsPage/index.js]: add a breakpoint to the code in the browser right there so you can play around w/ the location.history object to figure out how to extract the previous pathname from it. I can't recall off the top of my head.

File: app/routes.js [hunk 7]
```diff
@@ -36,7 +40,13 @@ export default (
       <Route path="verify" component={VerifyPage} />
     </Route>
     <Route path="/settings" component={SettingsPage} />
-    <Route path="/proofofstake" component={TicketsPage} />
+    <Route path="/tickets" component={TicketsPage}>
+      <IndexRedirect to="purchase" />
+      <Route path="purchase" component={TicketsPurchaseTab} position={0}/>
+      <Route path="mytickets" component={TicketsMyTicketsTab} position={1}/>
+      <Route path="governance" component={TicketsGovernanceTab} position={2}/>
+      <Route path="statistics" component={TicketsStatisticsTab} position={3}/>
```
Comment (Jascha-Sundaresan) [inline app/routes.js]: you don't need the position props

File: app/style/ImportScriptModal.less [hunk 8]
```diff
@@ -4,8 +4,10 @@
   position: absolute;
   z-index: 10;
   overflow: auto;
-  height: 556px;
-  padding: 54px 140px 54px 160px;
+  height: 100%;
+  padding: 54px 160px 54px 180px;
```
Comment (Jascha-Sundaresan) [inline app/style/ImportScriptModal.less]: semantically, I think you mean "margin" here, unless you're talking about _inside_ the modal.

**Comment by dajohi** [comment]
@Jascha-Sundaresan can you give this another review please?

**Comment by Jascha-Sundaresan** [comment]
checking now...

File: app/components/views/TicketsPage/GovernanceTab/Page.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+          options={configuredStakePools}
+          value={stakePool}
+          onChange={onChangeStakePool}
+        />
+      </div> :
+      <div></div>
+      }
+    </div>
+    <div className="stakepool-voting-agenda-area">
+      {selectedAgenda ? (
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/GovernanceTab/Page.js]: this is really minor, but we don't need to use the { query ? <Component/> : null } pattern anymore.

react 16 allows 'false' as a valued return for a component now, so you can just do { query && <Component/> }

File: app/components/views/TicketsPage/MyTicketsTab/Page.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/MyTicketsTab/Page.js]: `React` is being provided by webpack's providePlugin.  No need to include this anymore.

File: app/components/views/TicketsPage/PurchaseTab/index.js [hunk 11]
```diff
@@ -0,0 +1,153 @@
+import React, { Component } from "react";
+import { autobind } from "core-decorators";
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/PurchaseTab/index.js]: same with `autobind` this is being provided by webpack's providePlugin

File: app/components/views/TicketsPage/index.js [hunk 12]
```diff
@@ -1,174 +1,101 @@
...
-        }}
-      />
+  render () {
+    const { children, location } = this.props;
+    const { prevRoute } = this.state;
+    const tabs = ["purchase", "mytickets", "governance", "statistics"];
+    const page = "tickets";
+    // this will be removed w/ react router 4
+    const pathname = location.pathname.split("/")[2];
+    // Still unable to determine 'from' tab which would determine which transition to use for middle tabs
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/index.js]: this comment is no longer accurate

File: app/components/views/TicketsPage/index.js [hunk 13]
```diff
@@ -1,174 +1,101 @@
...
+      />
+    );
   }
 }
 
-export default service(ticketsPage(Tickets));
+Tickets.propTypes = propTypes;
+
+export default withTabSlide(tickets(Tickets));
+*/
```
Comment (Jascha-Sundaresan) [inline app/components/views/TicketsPage/index.js]: What's all this commented out stuff? Can it be deleted?
